### ISSUE TITLE
Improvement to autoscale.js service to remove Big-IPs after failed DSC join

### ISSUE THREAD

**Issue by megamattzilla** [F5Networks/f5-azure-arm-templates#211]
## Modify autoscale.js service to remove a new Big-IP from DSC if the first config sync fails.   
For the autoscale ARM templates - Modify autoscale.js service so that if after adding a new VMSS instance to existing DSC cluster and it fails to config sync successfully to receive existing cluster configuration, remove the failed VMSS instance from the DSC cluster.  

If you have an autoscale WAF ARM template deployed with multiple existing Big-IP devices, and VMSS spins up a new instance autoscale.js seems to be responsible for adding the new Big-IP to the existing DSC cluster and performing the first ever config sync to the new Big-IP.   

If for any reason the running configuration of the existing DSC cluster is not compatible with the new Big-IP, the config sync will fail and the new Big-IP will be left in the DSC cluster when autoscale.js eventually times out trying to add the new Big-IP to the DSC cluster.   

This leaves the config-sync (which is set to automatic sync by default) in an unstable state as any new change made on the new Big-IP (which never received the full config) can wipe the entire running DSC configuration with blank config. Ideally if a new Big-IP cannot receive the first ever config sync successfully it should be removed from DSC cluster when autoscale.js hits the timeout for new DSC join.   

A practical example of triggering this problem is if the Big-IP vnet and ltm pool member vnet share IP space. IP addresses of the LTM pools could eventually overlap with IPs addresses assigned to the Big-IP NIC when a new Big-IP is created.  
In this event, DSC sync to the new Big-IP will fail as it has an overlapping IP with the ltm pool. The new Big-IP will still be in the DSC cluster with the other existing nodes and automatic sync is stuck in changes pending (red). Any config change made on the new Big-IP device will overwrite all the existing DSC configurations.  

You can replicate this problem by having an existing DSC cluster created and configure LTM pool member IPs that you know could be assigned to additional Big-IP instances, then have VMSS create the additional VMSS Big-IP instance. After the new Big-IP fails to join, if you make any configuration changes on this device it will perform a config sync to the existing Big-IPs and erase their configuration. 

Example of autoscale.js service logs when a new Big-IP fails to join an existing DSC cluster and times out:  
autoscale.js log:
```
2020-09-10T03:19:39.975Z info:  Cluster action JOIN 
2020-09-10T03:19:39.976Z info:  Initializing encryption 
2020-09-10T03:19:39.976Z info:  Setting config sync IP. 
2020-09-10T03:19:40.100Z info:  2020-09-10T03:19:40.100Z info:  Joining cluster. 
2020-09-10T03:19:40.101Z info:  Joining cluster. 
2020-09-10T03:19:48.196Z info:  2020-09-10T03:19:48.196Z info:  This is a BIG-IP 
2020-09-10T03:19:48.197Z info:  2020-09-10T03:19:48.197Z info:  Waiting for device to be ready. 
2020-09-10T03:19:48.196Z info:  This is a BIG-IP 
2020-09-10T03:19:48.197Z info:  Waiting for device to be ready. 
2020-09-10T03:19:48.259Z info:  2020-09-10T03:19:48.259Z info:  Checking remote host for cluster readiness. 
2020-09-10T03:19:48.259Z info:  Checking remote host for cluster readiness. 
2020-09-10T03:19:48.346Z info:  Getting local hostname for trust. 
2020-09-10T03:19:48.347Z info:  2020-09-10T03:19:48.346Z info:  Getting local hostname for trust. 
2020-09-10T03:19:48.357Z info:  Getting local management address. 
2020-09-10T03:19:48.358Z info:  2020-09-10T03:19:48.357Z info:  Getting local management address. 
2020-09-10T03:19:48.366Z info:  Adding to remote trust. 
2020-09-10T03:19:48.367Z info:  2020-09-10T03:19:48.366Z info:  Adding to remote trust. 
2020-09-10T03:20:23.453Z info:  Adding to remote device group. 
2020-09-10T03:20:23.454Z info:  2020-09-10T03:20:23.453Z info:  Adding to remote device group. 
2020-09-10T03:20:27.269Z info:  Checking for datasync-global-dg. 
2020-09-10T03:20:27.270Z info:  2020-09-10T03:20:27.269Z info:  Checking for datasync-global-dg. 
2020-09-10T03:20:33.519Z info:  Telling remote to sync. 
2020-09-10T03:20:33.519Z info:  2020-09-10T03:20:33.518Z info:  Telling remote to sync. 
2020-09-10T03:21:11.416Z info:  Telling remote to sync datasync-global-dg request. 
2020-09-10T03:21:11.416Z info:  2020-09-10T03:21:11.416Z info:  Telling remote to sync datasync-global-dg request. 
2020-09-10T03:21:11.916Z info:  Waiting for sync to complete. 
2020-09-10T03:21:11.916Z info:  2020-09-10T03:21:11.916Z info:  Waiting for sync to complete. 
2020-09-10T03:21:46.139Z info:  Sync not yet complete. 
2020-09-10T03:21:46.139Z info:  2020-09-10T03:21:46.138Z info:  Sync not yet complete. 
2020-09-10T03:21:46.140Z info:  2020-09-10T03:21:46.139Z info:  Recommended action to sync group Sync 
2020-09-10T03:21:46.140Z info:  Recommended action to sync group Sync 
2020-09-10T03:22:51.319Z info:  Telling remote to sync. 
2020-09-10T03:22:51.319Z info:  2020-09-10T03:22:51.312Z info:  Telling remote to sync. 
2020-09-10T03:23:21.812Z info:  Telling remote to sync datasync-global-dg request. 
2020-09-10T03:23:21.812Z info:  2020-09-10T03:23:21.811Z info:  Telling remote to sync datasync-global-dg request. 
2020-09-10T03:23:21.946Z info:  Waiting for sync to complete. 
2020-09-10T03:23:21.947Z info:  2020-09-10T03:23:21.946Z info:  Waiting for sync to complete. 
2020-09-10T03:23:55.479Z info:  Sync not yet complete. 
2020-09-10T03:23:55.480Z info:  2020-09-10T03:23:55.479Z info:  Sync not yet complete. 
2020-09-10T03:23:55.481Z info:  2020-09-10T03:23:55.479Z info:  Recommended action to sync group Sync 
2020-09-10T03:23:55.480Z info:  Recommended action to sync group Sync 
2020-09-10T03:25:00.538Z info:  Telling remote to sync. 
2020-09-10T03:25:00.538Z info:  2020-09-10T03:25:00.538Z info:  Telling remote to sync. 
2020-09-10T03:25:30.730Z info:  Telling remote to sync datasync-global-dg request. 
2020-09-10T03:25:30.730Z info:  2020-09-10T03:25:30.728Z info:  Telling remote to sync datasync-global-dg request. 
2020-09-10T03:25:30.859Z info:  Waiting for sync to complete. 
2020-09-10T03:25:30.859Z info:  2020-09-10T03:25:30.859Z info:  Waiting for sync to complete.
```

LTM log shows the specific sync error: 
```
Sep 10 03:29:25 vmss10-vmss_16 err mcpd[5411]: 01071488:3: Remote transaction for device group /Common/Sync to commit id 81 6870695660948160877 /Common/vmss10-vmss_11.azuresecurity.com 0 failed with error 01070080:3: The requested pool member is already in use as a self IP address (10.7.1.13). 
```

Eventually autoscale.js will stop trying to perform the config-sync that keeps failing for the same reason and give up.  
When it gives up, it leaves the new Big-IP in the DSC group which can lead to sync happening in the wrong direction.  
If autoscale.js removed the problematic node from DSC cluster after first time config sync fails it would mitigate sync happening in the wrong direction.  

## Template
Template used: https://github.com/F5Networks/f5-azure-arm-templates/tree/v7.1.1.0/supported/autoscale/waf/via-lb/1nic/existing-stack/bigiq  
Although I think this behavior is controlled by cloud-libs. 

## Severity Level
Severity: 3

**Comment by shyawnkarim** [comment]
Thanks for reporting this enhancement. We are now tracking this request internally with ID ESECLDTPLT-2337.

**Comment by shyawnkarim** [comment]
Closing due to age. These legacy templates are now in maintenance mode and are being replaced by our next-generation templates available in the [Cloud Templates 2.0 GitHub repo](https://github.com/F5Networks/f5-azure-arm-templates-v2).
